### ISSUE TITLE
Job to renew subscriptions

### ISSUE THREAD

**Issue by russss** [UKHackspaceFoundation/website#120]
I believe we don't currently have the capability to capture membership payments after the first one. There needs to be a scheduled job which checks which memberships are due to expire and submits a new GoCardless payment for them.

Related to #52.

**Comment by qpnaosc** [comment]
Hello, it's Cleo. Does this involve any sensitive info (I.e. card details?) If that's been discussed before let me know, I'll look around a bit

**Comment by castaway** [comment]
> Hello, it's Cleo. Does this involve any sensitive info (I.e. card details?) If that's been discussed before let me know, I'll look around a bit

No, all the sensitive stuff is handled by GoCardless, we just need to write something that makes the GoCardless API retrieve more money from members via the direct debit system (GC handles that) presumably via a scheduled task / cronjob or similar

**Comment by proffalken** [comment]
I'm sure this was discussed on Telegram somewhere, but https://gocardless.com/solutions/subscription-payments/ means that we don't need a cron job, we just set up the payment and GoCardless handle all of that side of it as well.

I'm currently using the equivalent functionality of Stripe for Make Monmouth, and it works very well, no need for us to do anything on our side other than setup the initial payment details.

**Comment by castaway** [comment]
If that;s the case, what're we missing? We did capture initial payments from some folks, but the subsequent ones didnt happen - so we must be missing *something*

**Comment by proffalken** [comment]
@castaway - no idea tbh, I don't have access to the GoCardless account.

I know that when I was looking into it a few years back the subscription functionality was either missing or lacking (I forget which) so it could have been that it wasn't suitable last time it was looked at?

**Comment by proffalken** [comment]
Found the telegram chat:

```
yeah, I believe the reason we didn't use that was because for spaces we wanted the amount paid to be adjustable

and it wasn't possible to do that

(the gocardless subscription product is nice if you just want a standalone thing to take money from people - we use it for Limehouse Labs with no additional software)
```

So might be worth checking to see if we can vary the amounts per space?

FWIW, with Make Monmouth we just have two subscription levels (£5/month & £10/month) and users choose which one they want.

As long as we don't need to vary the amounts too much, perhaps GoCardless offers a similar kind of setup, and the space just selects the membership fee based on the number of members they have or some other criteria?

**Comment by russss** [comment]
You can amend subscription amounts in GoCardless now - you couldn't before. I think they added that feature relatively recently. So we could probably switch to using the subscription feature now.

**Comment by castaway** [comment]
Aha, a bonus from failing to make progress! ;)

**Comment by qpnaosc** [comment]
Seems like this one has been solved then :)

**Comment by proffalken** [comment]
> Seems like this one has been solved then :)

Heh, it hasn't been implemented yet though.

I tend to tell the teams I work with  `it's only "done" when the working software is in the hands of the user *and* monitoring/observability is in place` ;)

**Comment by proffalken** [comment]
EMF2024 - Who can do this? Because someone needs to :D
